### PR TITLE
feat: Convert components to typescript

### PR DESCRIPTION
**Pull request by JohanAludden**
Can I get some feedback and see if this is the way you want to solve the issue #22?

PR checklist:

- [x] Github Issue associated with branch.
- [x] Code passes eslint analysis.
- [ ] Code passes cross browser testing.
- [ ] Test code coverage is 85% or greater.
- [ ] PR branch is up-to-date with latest master branch.

### CODE REVIEW

File: src/FormattedMessage.tsx [hunk 1]
```diff
@@ -1,14 +1,19 @@
 import React from 'react';
-import PropTypes from 'prop-types';
 import { TextContext } from './TextProvider';
 
-const FormattedMessage = (props) => (
+interface FormattedMessageProps {
+  id: string;
+  values?: {};
```
Comment (MadaraUchiha-314) [inline src/FormattedMessage.tsx:6]: We can make `values` even stricter by allowing only `strings` as the key/value.

File: src/TextProvider.tsx [hunk 2]
```diff
@@ -0,0 +1,20 @@
+import React from 'react';
+
+export const TextContext = React.createContext<object>({});
+
+interface TextProviderProps {
+  globalText: any;
```
Comment (MadaraUchiha-314) [inline src/TextProvider.tsx:6]: `globalText` is an object of key/value as string/string. Can we add the definition here ?

**Review by MadaraUchiha-314 — COMMENTED** [review]
There would be changes required to the build, linter etc. apart from the code changes to convert it to a typescript component.

File: src/FormattedMessage.tsx (hunk 1 above)
Comment (JohanAludden) [reply src/FormattedMessage.tsx]: yes, changed it to `values?: { [key: string]: string };`

File: src/TextProvider.tsx (hunk 2 above)
Comment (JohanAludden) [reply src/TextProvider.tsx]: yes, changed it to `globalText: { [key: string]: string };`

File: .eslintrc.json [hunk 3]
```diff
@@ -1,21 +1,13 @@
...
+  "root": true,
+  "parser": "babel-eslint",
+  "extends": ["eslint:recommended", "plugin:react/recommended", "airbnb"],
+  "settings": {
+    "import/extensions": [".js", ".jsx", ".ts", ".tsx"]
+  },
+  "rules": {
+    "import/extensions": "off",
+    "import/no-unresolved": "off",
+    "react/jsx-filename-extension": [2, { "extensions": [".jsx", ".tsx", ".ts"] }]
```
Comment (MadaraUchiha-314) [inline .eslintrc.json:11]: Why is `.ts` there in `react/jsx-filename-extension` ?

**Review by MadaraUchiha-314 — CHANGES_REQUESTED** [review]
@JohanAludden Except for one comment and one dummy file, everything looks good to go!

**Comment by JohanAludden** [comment]
Made the requested changes. 
Just want to point out that when you do not have a `.ts` file, you can not have it as part of the eslint script in the `package.json` file.

**Review by MadaraUchiha-314 — APPROVED** [review]
LGTM.

**Pull request closed without merging**